src/test/java/tests/JavaVersionTest.java

```diff
@@ -0,0 +1,135 @@
+package tests;
+
+import org.junit.Test;
+import util.JavaVersion;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.fail;
+
+public class JavaVersionTest {
+    @Test
+    public void javaVersionParsing_Java8VersionNoBuild_ParsedVersionCorrectly() {
+        String version = "1.8.0_60";
+        JavaVersion expectedVersion = new JavaVersion(1, 8, 0, 60, 0);
+        JavaVersion javaVersion = JavaVersion.fromString(version);
+        assertEquals(expectedVersion, javaVersion);
+    }
+
+    @Test
+    public void javaVersionParsing_Java8VersionWithBuild_ParsedVersionCorrectly() {
+        String version = "1.8.0_60-b40";
+        JavaVersion expectedVersion = new JavaVersion(1, 8, 0, 60, 40);
+        JavaVersion javaVersion = JavaVersion.fromString(version);
+        assertEquals(expectedVersion, javaVersion);
+    }
+
+    @Test
+    public void javaVersionParsing_Java8VersionBigNumbers_ParsedVersionCorrectly() {
+        String version = "10.80.100_60";
+        JavaVersion expectedVersion = new JavaVersion(10, 80, 100, 60, 0);
+        JavaVersion javaVersion = JavaVersion.fromString(version);
+        assertEquals(expectedVersion, javaVersion);
+
+        version = "010.080.100_60";
+        expectedVersion = new JavaVersion(10, 80, 100, 60, 0);
+        javaVersion = JavaVersion.fromString(version);
+        assertEquals(expectedVersion, javaVersion);
+    }
```

Comment:
Looks like you can merge the three tests above into one `javaVersion_acceptableVerionString` or something like that. Furthermore, you can extract a utility method `verifyVersionParsedCorrectly(String, int, int, int, int, int)`